File: 1281.cpp
// Subtract the product and sum of digits of a number

// Runtime : 0ms ; Memory : 6MB

class Solution {
public:
    int subtractProductAndSum(int n) {
        int product = 1, sum = 0, remainder = 0;
        while(n!=0) {
            remainder = n % 10;
            product = product * remainder;
            sum = sum + remainder;
            n = n / 10;
        }
        return (product-sum);
    }
};

File: 1281.py
# Subtract the product and sum of digits of a number

# Runtime : 35ms ; Memory : 13.9 MB

class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        su = 0
        product = 1
        while n!= 0:
            remainder = n % 10
            su = su + remainder
            product = product * remainder
            n = n//10
        return product - su


# Solution 2 : Space = O(log n)

def subtractProductAndSum(self, n):
        A = map(int, str(n))
        return reduce(operator.mul, A) - sum(A)